src/cow-react/modules/limitOrders/containers/LimitOrdersWidget/index.tsx

```diff
@@ -111,9 +113,15 @@ export function LimitOrdersWidget() {
   const onSwitchTokens = useCallback(() => {
     const { inputCurrencyId, outputCurrencyId } = state
 
-    updateLimitOrdersState({ inputCurrencyAmount: outputCurrencyAmount?.toExact(), outputCurrencyAmount: null })
+    updateLimitOrdersState({
+      inputCurrencyAmount: outputCurrencyAmount?.toExact(),
```

Comment:
As discussed today in the meeting, should we update this logic to always keep the value of the field where the user typed value and then move that value to another field on switch and then recalculate the value of the other field based on price/rate?

I think right now in limit orders state we don't keep track of independent field, but we can get that info based on the current order type. If its sell order then input is independent and if its buy order type then its output. So we keep values of those fields.

Reply:
Yes, @elena-zh created an issue for this